webidl/Generate.hx

```diff
@@ -99,10 +88,19 @@ template<typename T> pref<T> *_alloc_const( const T *value ) {
 		add(StringTools.trim(gc ? HEADER_GC : HEADER_NO_GC));
 		add("");
 		add("#endif");
-		if( opts.includeCode != null ) {
+
+		/* Handle Includes TODO@Wolfie -> Dedup? **/
+		for(sFile in opts.sourceFiles){
 			add("");
-			add(StringTools.trim(opts.includeCode));
+			for(l in sys.io.File.getBytes(if(sys.FileSystem.exists(opts.out+ "/" + sFile)){

```

Comment:
I find this block of code very difficult to read. This is a bit cleaner:


```haxe
var path = null;
for(sFile in opts.sourceFiles){
	add("");
	path = FileSystem.exists(opts.out + "/" + sFile) ? opts.out + "/" + sFile : sFile;
	for(l in sys.io.File.getContent(path).split("\n")){
		if(StringTools.startsWith(l, "#include")){
			add(StringTools.trim(l));
		}
	}
}
```

Also for large projects would this be a performance concern, going through all of the source files that could represent a large library? I don't know how long that takes, it probably isn't a problem, but I thought I'd bring it up. Maybe we should test it with a bunch of files or add a `autoIncludeIncludes` ( :wink: ) flag or something like that to the `webidl.json` ( maybe with a better name like `detectIncludes` ).

Reply:
Looks good to me.. I'm coming from Kotlin and I've been embracing it's functional side so I've become used to long functions.. however you want it man is fine with me.